### DISCUSSION TITLE
start chrome with user profile

### DISCUSSION THREAD

**Discussion by Giorgio-Vi** [jef/streetmerchant#2347] [Help]
now starting chrome with "PUPPETEER_EXECUTABLE_PATH =" C: \ Program Files \ Google \ Chrome \ Application \ chrome.exe "" chrome is free from any addon.
is it possible start it with your own user version?
This to manage cookies in a personalized way for each site and start extensions such as nordvpn for chrome.

Sorry for the incorrect English.

**Comment by jef** [comment]
Use `node ./node_modules/puppeteer/install.js` and don't set those environment variables. That should resolve your issues.

**Reply by Giorgio-Vi** [reply]
What do you mean for "don't set those environment variables"?
I installed /node_modules/puppeteer/install.js and installed chromium, if deactivate "PUPPETEER_EXECUTABLE_PATH" anyway a clean version of chromium is started.
did I do something wrong?

**Reply by jef** [reply]
Oh, I'm sorry. I misread your title. You want to use a profile that you already have in Chrome, correct?

**Reply by jef** [reply]
You might have to give double backslashes, but I could be wrong. I can't test on a Windows machine currently.

`PUPPETEER_EXECUTABLE_PATH="C:\\Program Files\\Google Chrome\\Application\\chrome.exe"`

**Reply by Giorgio-Vi** [reply]
> You might have to give double backslashes, but I could be wrong. I can't test on a Windows machine currently.
> 
> `PUPPETEER_EXECUTABLE_PATH="C:\\Program Files\\Google Chrome\\Application\\chrome.exe"`

no, with double backslashes there is no difference.

**Reply by Giorgio-Vi** [reply]
> Oh, I'm sorry. I misread your title. You want to use a profile that you already have in Chrome, correct?

Correct. It doesn't need to be chrome, chromium is fine too. just start with your custom profile and extensions.

I found this article interesting
https://dev.to/ajaykumbhare/load-chrome-extensions-in-puppeteer-4fk0

if I didn't understand it would be enough to set "ignoreDefaultArgs" to true, but I have no idea how to set it to streetmerchant.

I take this opportunity to compliment you on everything.

**Reply by jef** [reply]
So I tried a variance of things. I tried what you posted above, I've tried using `userDataDir`, tried profiles, tried launch a websocket connection -- all of which didn't work. Although the userDataDir one did work with profiles, but not what was mine.

If you'd like to try yourself, go to chrome://version, find your user profile and executable, then add it to https://github.com/jef/streetmerchant/blob/e2d804027a3ec6a19201ef9ce35dcb4c4dee7025/src/index.ts#L62-L69

So it will look something like this afterward:

```typescript
  browser = await launch({
    args,
    defaultViewport: {
      height: config.page.height,
      width: config.page.width,
    },
    executablePath: '/usr/bin/google-chrome-stable',
    userDataDir: '/home/username/.config/google-chrome/Default',
    headless: config.browser.isHeadless,
  });
```

This is Linux, so that paths are a little different, but you get jist. Let me know how it works for you! And if it doesn't, I'm sorry I can't be of any more help!

**Reply by Giorgio-Vi** [reply]
> So I tried a variance of things. I tried what you posted above, I've tried using `userDataDir`, tried profiles, tried launch a websocket connection -- all of which didn't work. Although the userDataDir one did work with profiles, but not what was mine.
> 
> If you'd like to try yourself, go to chrome://version, find your user profile and executable, then add it to
> 
> https://github.com/jef/streetmerchant/blob/e2d804027a3ec6a19201ef9ce35dcb4c4dee7025/src/index.ts#L62-L69
> 
> So it will look something like this afterward:
> 
> ```ts
>   browser = await launch({
>     args,
>     defaultViewport: {
>       height: config.page.height,
>       width: config.page.width,
>     },
>     executablePath: '/usr/bin/google-chrome-stable',
>     userDataDir: '/home/username/.config/google-chrome/Default',
>     headless: config.browser.isHeadless,
>   });
> ```
> 
> This is Linux, so that paths are a little different, but you get jist. Let me know how it works for you! And if it doesn't, I'm sorry I can't be of any more help!

it doesn't work in windows, but it's the right approach.
I understand you have to load the necessary "angs" anyway (for example one of those I need is --load-extension, etc.) but I don't know how to integrate it.

**Comment by WallisonLima** [comment]
Olá, você pode experimentar essa função que criei: 

`async function openChrome(profile){

	return new Promise(async (resolve, reject) => {

		let fkr = `google-chrome --profile-directory="`+k[profile]+`" --remote-debugging-port=9222`

		exc(fkr)

		console.log('started')

		await sleep(6000)

		request({url: 'http://127.0.0.1:9222/json/version', method: 'GET'}, 

        async function(e, b, lbody){

            resolve(JSON.parse(lbody).webSocketDebuggerUrl)

    	           })	
	})
}`

o parâmentro k deve ser algo parecido com isso 

`k["P1"] = "Profile 1"` onde `Profile 1` é o nome do perfil da conta do navegador.

Após seu chamado ela cria a variável `fkr` com parametros passados a `exc`... que deve ser baixado como dependência com o comando 

`npm install child_process` 

ou caso de errado com esta instalação, de uma olhada no repositório https://nodejs.org/api/child_process.html.

Após isso baixe também a dependência do request:

`npm install request`

feito isso só declare as constantes:

`const request = require('request')
const exc = require('child_process').exec`

Espero que ajude!
